File: InventoryManagementSystem/wwwroot/js/Items/Item.Model.js
/// <reference path="../knockout.js" />

var ItemModel = function (item) {
    var self = this;
    item = item || {};
    self.Id = ko.observable(item.id || 0); 
    self.Name = ko.observable(item.name || '');
    self.Unit = ko.observable(item.unit || '');
    self.Category = ko.observable(item.category || '');
}

File: Inventory.Entities/ItemModel.cs
using System.ComponentModel.DataAnnotations;

namespace InventoryManagementSystem.Models
{
    public class ItemModel
    {
        [Key]
        public int Id { get; set; }
        [Required]
        [StringLength(25, ErrorMessage ="Name can't be greater than 25 character")]
        [Display(Name = "Item Name")]
        public string Name { get; set; }
        [Required]
        [Display(Name = "Unit")]
        public string Unit { get; set; }
        [Required]
        [Display(Name = "Category")]
        public string Category { get; set; }
    }
}
